Should Ops own tests pass with both SCENARIO_BARE_CHARM_ERRORS on and off?

```command
> pytest test/test_model_relation_data_class.py
...
=== 54 passed ...
```

```command
> env SCENARIO_BARE_CHARM_ERRORS=1 pytest test/test_model_relation_data_class.py
...
=== 16 failed, 38 passed ...
```

The failing cases all stem from the same source:
```
FAILED test/test_model_relation_data_class.py::test_relation_load_fail[MyCharm] - ValueError: bar must be a zero or positive int
FAILED test/test_model_relation_data_class.py::test_relation_load_fail[MyDataclassCharm] - ValueError: bar must be a zero or positive int
FAILED test/test_model_relation_data_class.py::test_relation_load_fail[MyPydanticDataclassCharm] - pydantic_core._pydantic_core.ValidationError: 1 validation error for MyPydanticDataclassDatabag
FAILED test/test_model_relation_data_class.py::test_relation_load_fail[MyPydanticBaseModelCharm] - pydantic_core._pydantic_core.ValidationError: 1 validation error for MyPydanticDatabag
FAILED test/test_model_relation_data_class.py::test_relation_load_fail_multi_field_validation[MyCharm] - ValueError: bar must be a zero or positive int
FAILED test/test_model_relation_data_class.py::test_relation_load_fail_multi_field_validation[MyDataclassCharm] - ValueError: bar must be a zero or positive int
FAILED test/test_model_relation_data_class.py::test_relation_load_fail_multi_field_validation[MyPydanticDataclassCharm] - pydantic_core._pydantic_core.ValidationError: 1 validation error for MyPydanticDataclassDatabag
FAILED test/test_model_relation_data_class.py::test_relation_load_fail_multi_field_validation[MyPydanticBaseModelCharm] - pydantic_core._pydantic_core.ValidationError: 1 validation error for MyPydanticDatabag
FAILED test/test_model_relation_data_class.py::test_relation_save_no_access[MyCharm] - ops.model.RelationDataAccessError: foo cannot write the data of remote application remote
FAILED test/test_model_relation_data_class.py::test_relation_save_no_access[MyDataclassCharm] - ops.model.RelationDataAccessError: foo cannot write the data of remote application remote
FAILED test/test_model_relation_data_class.py::test_relation_save_no_access[MyPydanticDataclassCharm] - ops.model.RelationDataAccessError: foo cannot write the data of remote application remote
FAILED test/test_model_relation_data_class.py::test_relation_save_no_access[MyPydanticBaseModelCharm] - ops.model.RelationDataAccessError: foo cannot write the data of remote application remote
FAILED test/test_model_relation_data_class.py::test_relation_save_invalid[MyCharm] - ops.model.RelationDataTypeError: relation data values must be strings, not <class 'int'>
FAILED test/test_model_relation_data_class.py::test_relation_save_invalid[MyDataclassCharm] - ops.model.RelationDataTypeError: relation data values must be strings, not <class 'int'>
FAILED test/test_model_relation_data_class.py::test_relation_save_invalid[MyPydanticDataclassCharm] - ops.model.RelationDataTypeError: relation data values must be strings, not <class 'int'>
FAILED test/test_model_relation_data_class.py::test_relation_save_invalid[MyPydanticBaseModelCharm] - ops.model.RelationDataTypeError: relation data values must be strings, not <class 'int'>
```

--- a/test/test_model_relation_data_class.py
+++ b/test/test_model_relation_data_class.py
@@ -241,7 +241,7 @@ def test_relation_load_simple(charm_class: type[BaseTestCharm]):
 
 @pytest.mark.parametrize('charm_class', _test_classes)
 def test_relation_load_fail(charm_class: type[BaseTestCharm], monkeypatch: pytest.MonkeyPatch):
-    monkeypatch.setenv('SCENARIO_BARE_CHARM_ERRORS', 'false')
+    monkeypatch.setenv('SCENARIO_BARE_CHARM_ERRORS', 'true')
 
     class Charm(charm_class):
         def _on_relation_changed(self, event: ops.RelationChangedEvent):
@@ -252,16 +252,15 @@ def test_relation_load_fail(charm_class: type[BaseTestCharm], monkeypatch: pytes
     data = {'foo': json.dumps('value'), 'bar': json.dumps('bar'), 'baz': json.dumps(['a', 'b'])}
     rel = testing.Relation('db', remote_app_data=data)
     state_in = testing.State(leader=True, relations={rel})
-    with pytest.raises(testing.errors.UncaughtCharmError) as exc_info:
+    with pytest.raises(ValueError):
         ctx.run(ctx.on.relation_changed(rel), state_in)
-    assert isinstance(exc_info.value.__cause__, ValueError)
 
 
 @pytest.mark.parametrize('charm_class', _test_classes)
 def test_relation_load_fail_multi_field_validation(
     charm_class: type[BaseTestCharm], monkeypatch: pytest.MonkeyPatch
 ):
-    monkeypatch.setenv('SCENARIO_BARE_CHARM_ERRORS', 'false')
+    monkeypatch.setenv('SCENARIO_BARE_CHARM_ERRORS', 'true')
 
     class Charm(charm_class):
         def _on_relation_changed(self, event: ops.RelationChangedEvent):
@@ -276,9 +275,8 @@ def test_relation_load_fail_multi_field_validation(
     }
     rel = testing.Relation('db', remote_app_data=data)
     state_in = testing.State(leader=True, relations={rel})
-    with pytest.raises(testing.errors.UncaughtCharmError) as exc_info:
+    with pytest.raises(ValueError):
         ctx.run(ctx.on.relation_changed(rel), state_in)
-    assert isinstance(exc_info.value.__cause__, ValueError)
 
 
 class _AliasProtocol(Protocol):
@@ -427,7 +425,7 @@ def test_relation_save_simple(charm_class: type[BaseTestCharm]):
 def test_relation_save_no_access(
     charm_class: type[BaseTestCharm], monkeypatch: pytest.MonkeyPatch
 ):
-    monkeypatch.setenv('SCENARIO_BARE_CHARM_ERRORS', 'false')
+    monkeypatch.setenv('SCENARIO_BARE_CHARM_ERRORS', 'true')
 
     class Charm(charm_class):
         def _on_relation_changed(self, event: ops.RelationChangedEvent):
@@ -437,9 +435,8 @@ def test_relation_save_no_access(
     ctx = testing.Context(Charm, meta={'name': 'foo', 'requires': {'db': {'interface': 'db-int'}}})
     rel_in = testing.Relation('db')
     state_in = testing.State(leader=True, relations={rel_in})
-    with pytest.raises(testing.errors.UncaughtCharmError) as exc_info:
+    with pytest.raises(ops.RelationDataAccessError):
         ctx.run(ctx.on.relation_changed(rel_in), state_in)
-    assert isinstance(exc_info.value.__cause__, ops.RelationDataAccessError)
 
 
 @pytest.mark.parametrize('charm_class', _test_classes)
@@ -474,7 +471,7 @@ def test_relation_load_then_save(charm_class: type[BaseTestCharm]):
 
 @pytest.mark.parametrize('charm_class', _test_classes)
 def test_relation_save_invalid(charm_class: type[BaseTestCharm], monkeypatch: pytest.MonkeyPatch):
-    monkeypatch.setenv('SCENARIO_BARE_CHARM_ERRORS', 'false')
+    monkeypatch.setenv('SCENARIO_BARE_CHARM_ERRORS', 'true')
 
     class Charm(charm_class):
         def _on_relation_changed(self, event: ops.RelationChangedEvent):
@@ -488,9 +485,8 @@ def test_relation_save_invalid(charm_class: type[BaseTestCharm], monkeypatch: py
     ctx = testing.Context(Charm, meta={'name': 'foo', 'requires': {'db': {'interface': 'db-int'}}})
     rel_in = testing.Relation('db')
     state_in = testing.State(leader=True, relations={rel_in})
-    with pytest.raises(testing.errors.UncaughtCharmError) as exc_info:
+    with pytest.raises(ops.RelationDataTypeError):
         ctx.run(ctx.on.relation_changed(rel_in), state_in)
-    assert isinstance(exc_info.value.__cause__, ops.RelationDataTypeError)
 
 
 class _OneStringProtocol(Protocol):

